pkg/resource/deploy/step_generator.go

```diff
@@ -802,24 +743,27 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, res
 		if !isTargeted {
 			logging.V(7).Infof(
 				"Planner decided not to update '%v' due to not being in target group (same) (inputs=%v)", urn, new.Inputs)
-		} else {
-			updateSteps, res := sg.generateStepsFromDiff(
-				event, urn, old, new, oldInputs, oldOutputs, inputs, prov, goal, randomSeed)
+			// No need to update anything, the properties didn't change.
+			sg.sames[urn] = true
+			return []Step{NewSameStep(sg.deployment, event, old, new)}, nil
+		}
```

Comment:
isTargeted shows up below due to a short-circuit, maybe should flip the logic here to handle isTargeted (same vs skippedCreate)